Shell fixes

mikehearn opened the pull request.

chrisr3 commented:
Still getting this error when running from JRE:
```
Wed Mar 29 17:31:10 BST 2017>>> help
help: exception: NullPointerException
```

mikehearn commented:
Good catch. It turns out the jmx command is also implemented using JITC java. I've committed a changed fix that drops the Java compiler plugin entirely, which cleanly disables all such commands including any added in future. I also renamed tools.jar in my JDK and it seems to work. So can you try again?

chrisr3 commented:
Yes, the CRaSH `help` command no longer NPEs :-).

mikehearn commented:
So can you approve?

mikehearn commented on node/src/main/kotlin/net/corda/node/Corda.kt:
```diff
@@ -133,7 +134,11 @@ fun main(args: Array<String>) {
-                InteractiveShell.startShell(dir, runShell, cmdlineOptions.sshdServer, node)
+                try {
+                    InteractiveShell.startShell(dir, runShell, cmdlineOptions.sshdServer, node)
+                } catch(e: Throwable) {
```
Yes. I originally had Exception but I wasted a lot of time trying to figure out why it wasn't working only to discover that there was a NoClassDefFoundError being thrown. Catching and logging Throwable is fine.

chrisr3 replied:
It looks as if the code-base catches a _lot_ of `Throwable` exceptions; it's not pretty but Kotlin doesn't seem to support something like Java 7's `catch (ThisException | ThatException e)` syntax yet.
